src/color.rs

```diff
@@ -150,335 +97,606 @@ impl<'de> Deserialize<'de> for RGBA {
         D: Deserializer<'de>,
     {
         let (r, g, b, a) = Deserialize::deserialize(deserializer)?;
-        Ok(RGBA::new(r, g, b, a))
+        Ok(Self::new(r, g, b, a))
     }
 }
 
-impl ToCss for RGBA {
-    fn to_css<W>(&self, dest: &mut W) -> fmt::Result
-    where
-        W: fmt::Write,
-    {
-        let has_alpha = self.alpha != OPAQUE;
+// NOTE: `RGBA` should not implement `ToCss` because it is an internal primitive that does not directly represent a CSS value.
 
-        dest.write_str(if has_alpha { "rgba(" } else { "rgb(" })?;
-        self.red.to_css(dest)?;
-        dest.write_str(", ")?;
-        self.green.to_css(dest)?;
-        dest.write_str(", ")?;
-        self.blue.to_css(dest)?;
+#[inline]
+fn f32_to_f64(f: f32) -> f64 {
+    const PRECISION: f64 = 1e6;
+    (f64::from(f) * PRECISION).round() / PRECISION
+}
 
-        serialize_alpha(dest, self.alpha, true)?;
+/// <hue>
+/// https://w3c.github.io/csswg-drafts/css-color-4/#hue-syntax
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub struct Hue {
```

Comment:
maybe ```
type Degrees = f32;
struct Hue(Degrees);
```

Allows for better syntax like `Hue(32.9)`, etc.

Reply:
I intentionally implemented it the way I did to serve two purposes:
* Avoid having to use `.0` to get at the value.
* Make clear that the value has been normalized from what may have been originally specified (e.g. radians).

So needing to use `Hue::new(32.9)` was an intentional design decision.

I don't know if that changes your opinion?